cpp/src/arrow/engine/substrait/options.cc

```diff
@@ -205,12 +205,17 @@ class DefaultExtensionProvider : public BaseExtensionProvider {
       ARROW_ASSIGN_OR_RAISE(auto aggregate, internal::ParseAggregateMeasure(
                                                 agg_measure, ext_set, conv_opts,
                                                 /*is_hash=*/!keys.empty(), input_schema));
+      aggregate.name = aggregate.function;
```

Comment:
Why do we need this logic?

Reply:
Without this logic, the aggregate schema ends up with an empty name for each aggregate. This is because the [aggregate decoding](https://github.com/apache/arrow/blob/16dbd98e8f525884649e5a33096c74a455f64610/cpp/src/arrow/engine/substrait/extension_set.cc#L929) is [returning an aggregate with an empty name](https://github.com/apache/arrow/blob/16dbd98e8f525884649e5a33096c74a455f64610/cpp/src/arrow/engine/substrait/extension_set.cc#L980-L981), which in turn is likely because it has no access to the input schema against which to resolve the field reference (`*arg_ref`). The code here does have access to this schema and resolves the field reference.

Reply:
> Without this logic, the aggregate schema ends up with an empty name for each aggregate

While empty name is not great. I wonder if it actually breaks anything - substrait doesn't use names to refer to intermediate columns so IIUC the names here doesn't matter for correctness. (Although I agree meanful names are better for debug).

Does any code breaks without this change?

Reply:
I think you're right and this is mostly useful in debugging. However, note that the schema produced by the pre-PR code is inconsistent - aggregate columns have empty names while other column have "normal" names (derived from the input). So, this change make sense at least for consistency.

> Does any code breaks without this change?

No. However, after adding this logic I needed to change one test case that checks the aggregate column name. There must be a reason for the existence of this test case, but I don't know it. @westonpace ?

Reply:
@rtpsw I am happy to merge the code that fixes the the segmented aggregation bug (didn't pass the segmented key). However for the internal aggregation names change we would need validate that this naming is consistent with other aggregation codepath and/or refactor the naming logic to be shared (e.g., move that logic into ParseAggregateMeasure). I would suggest revert that change for now so we can move forward with the bug fix faster.

Reply:
Done.